docs/source/trainer_cli.rst

```diff
@@ -0,0 +1,156 @@
+simple as:
+
+.. code-block:: python
+
+    from pytorch_lightning.utilities.jsonargparse_utils import trainer_cli
+    from mycode import LitModel
+
+    trainer_cli(LitModel)
+
+The help of the tool describing all configurable options and default values can
+be shown by running :code:`python trainer.py --help`. Default options can be
+changed by providing individual command line arguments. However, it is better
+practice to create a configuration file and provide this to the trainer. A way
+to do this would be:
+
+.. code-block:: bash
+
+    # Dump default configuration to have as reference
+    python trainer.py --print-config > default_config.yaml
+    # Create config including only options to modify
+    nano config.yaml
+    # Run training using created configuration
+    python trainer.py --cfg config.yaml
+
+The call to the :func:`trainer_cli` function takes care of parsing command line
+and config file options, instantiating the classes, setting up a callback to
+save the config in the log directory and finally running :func:`trainer.fit`.
+
+After multiple trainings with different configurations, a previous run can be
+trivially reproduced by using the config in the respective log directory, e.g.:
+
+.. code-block:: bash
+
+    python trainer.py --cfg lightning_logs/version_7/config.yaml
+
+The start of a possible implementation of :class:`LitModel` including the
+recommended argument descriptions in the docstring could be the one below. Note
+that by using type hints and docstrings there is no need to duplicate this
+information to define its configurable arguments.
+
+.. code-block:: python
+
+    class LitModel(LightningModule):
+
+        def __init__(self,
+                     encoder_layers: int = 12,
+                     decoder_layers: List[int] = [2, 4]):
+            """Example encoder-decoder model
+
+            Args:
+                encoder_layers: Number of layers for the encoder
+                decoder_layers: Number of layers for each decoder block
+            """
+            ...
+
+If a separate :class:`LightningDataModule` class is required, the trainer tool
+just needs a small modification as follows:
+
+.. code-block:: python
+
+    from pytorch_lightning.utilities.jsonargparse_utils import trainer_cli
+    from mycode import LitModel, LitDataModule
+
+    trainer_cli(LitModel, LitDataModule)
+
+
+LightningArgumentParser
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Even though :func:`trainer_cli` can reduce boilerplate code to a minimum,
+clearly there are cases in which it is not enough. For this Lightning provides
+the :class:`LightningArgumentParser` class which is an extension of the built-in
+Python ArgumentParser that makes it very simple to implement configurable
+training tools with the same features as :func:`trainer_cli`.
+
+An example of a more complex training tool could be one in which there are
+several independent modules that require configuration. The code for such a case
+could look something like:
```

Comment:
Can you add an example of how the `--help` of the following snippet would look like?. Is each module inside their own ArgumentGroup by default?

Reply:
Yes, each module would have its own group. I will add how the help looks like.

Reply:
Added help snippet.